`datafusion-spark` fails to build standalone with `-p datafusion-spark`

`make_valid_utf8.rs` imports `datafusion::logical_expr::{...}` directly. `datafusion` is an optional dep of `datafusion-spark` gated behind the `core` feature, so:

```
$ cargo check -p datafusion-spark
error[E0433]: failed to resolve: use of unresolved module or unlinked crate `datafusion`
  --> datafusion/spark/src/function/string/make_valid_utf8.rs:20
```

Same failure for `cargo bench -p datafusion-spark`, which blocks running the spark criterion benches in isolation.

CI passes because `datafusion-sqllogictest` depends on `datafusion-spark` with `features = ["core"]`, so under `--workspace` feature unification `core` is always on. The failure only shows up when iterating on the spark crate by itself.

Fix: change the import to `datafusion_expr::{...}`, already imported a few lines below.

--- a/datafusion/spark/src/function/string/make_valid_utf8.rs
+++ b/datafusion/spark/src/function/string/make_valid_utf8.rs
@@ -17,13 +17,15 @@
 
 use arrow::array::{ArrayRef, LargeStringArray, StringArray};
 use arrow::datatypes::{DataType, Field, FieldRef};
-use datafusion::logical_expr::{ColumnarValue, Signature, Volatility};
 use datafusion_common::cast::{
     as_binary_array, as_binary_view_array, as_large_binary_array,
 };
 use datafusion_common::utils::take_function_args;
 use datafusion_common::{Result, internal_err};
-use datafusion_expr::{ReturnFieldArgs, ScalarFunctionArgs, ScalarUDFImpl};
+use datafusion_expr::{
+    ColumnarValue, ReturnFieldArgs, ScalarFunctionArgs, ScalarUDFImpl, Signature,
+    Volatility,
+};
 use datafusion_functions::utils::make_scalar_function;
 use std::sync::Arc;
 

